enhancements/metadata/user-tag-management.md

```diff
@@ -0,0 +1,396 @@
+7. Add/update tag workflow is initiated.
+
+#### Add/Update tags
+
+1. User creates `AWSMetadata` object with tag list details.
+2. User creates `CloudMetadata` configuration with controller configuration.
+3. Controller updates ready condition to false in `MetadataStatus.status`.
+4. Controller validates tag entries from `AWSMetadata.spec.resourcetags` and populates a new list of tags to be added to cloud resource.
+5. Controller appends tags to existing list and trims as per maximum limit policy specific to cloud resource.
+6. Controller performs lexicographic sorting on the trimmed tag list.
+7. Controller adds/updates the tag list on the cloud resource.
+8. Controller updates the tag list to `AWSMetadata.status`.
+9. Controller updates ready condition to true in `MetadataStatus.status`.
+
+#### Delete tags
+
+1. User adds tag key list (optionally, value, for validation) to custom resource specification.
+2. User specifies operation to be performed on the tag list by specific opcode.
+3. Controller updates ready condition to false in `MetadataStatus.status`.
+4. Controller gets the active tag list, removes requested tags from the list and populates a new list of tags.
+5. Controller replaces the tag list on the cloud resource.
+6. Controller updates the tag list to `AWSMetadata.status`.
+7. Controller updates ready condition to true in `MetadataStatus.status`.
+
+#### Handling of pre-existing tags or tags added using other tools
+
+In case of add operation, controller appends tags to existing tag list. If there is any tag pre-existing with same key name,
+there will be an error reported for the operation. While, in case of update operation, the pre-existing tag is updated. There is
+no distinction made between pre-existing tags and tags in `AWSMetadataSpec` for reconciliation. The active tag list is listed in the status.
+
+#### Classifier tag behaviour
+
+Classifier tags are used to identify cloud resources for which the controller should manage the tags. Classifier tags are mandatory and should
+be added to cloud resource. Controller does not support any wildcard characters in classifier strings.
+When multiple classifiers are used, a logical OR condition is applied for the classifiers.
+
+#### Reconciliation
+
+Reconciliation of tags is based on active tag list in `AWSMetadata.status.resourcetags`. Reconciliation of tags is ignored when ready condition is set to false at `CloudMetadata.MetadataStatus.status`. On deletion of AWSMetadataRef
+there is no reconciliation of tags. If AWSMetadataRef refers to non-existing `AWSMetadata` object , controller fails with error condition and updates `CloudMetadata.MetadataStatus.status` to false.
+
+#### No cloud provider configured
+
+When there is no cloud provider object configured, there will be no operation by default.
+
+#### Deleting classifier tag
+
+Controller does not delete, update or reconcile classifier tags. Controller refers them as read-only tags. 
```

Comment:
What is the impact of using a classifier tag, since this now uses up 1 out of the limited number of tags on a cloud, eg, IIRC S3 buckets can only have 10 tags, using one as a classifier is limiting that further, has there been any investigation into what the impact might be?